manifest-build-tools/application/update_rackhd_version.py

```diff
@@ -0,0 +1,207 @@
+                            else:
+                                depends_str = "         {0} (= {1}){2}".format(package_name, version_dict[package_name], os.linesep)
+                            new_control_fp.write(depends_str)
+                        else:
+                            new_control_fp.write("{0}{1}".format(package, os.linesep))
+                else:
+                    if not is_depends or package_count >= len(packages):
+                        new_control_fp.write(line)
+                    else:
+                        package_count += 1
+
+        new_control_fp.close()
+        os.remove(control)
+        os.rename(new_control, control)
+
+    def _generate_version_dict(self):
+        """
+        generate a dictory which includes the version of package on-xxx
+        :return: a dictory
+        """
+        version_dict = {}
+        for repo in os.listdir(self._builddir):
+            repo_dir = os.path.join(self._builddir, repo)
+            version_generator = VersionGenerator(repo_dir, self._manifest_repo_dir)
+            version = version_generator.generate_package_version(self._is_official_release)
+            if version != None:
+                version_dict[repo] = version
+
+        return version_dict
+
+    def update_RackHD_control(self):
+        """
+        udpate RackHD/debian/control according to manifest
+        :return: None     
+        """
+        try:
+            rackhd_dir = os.path.join(self._builddir, "RackHD")
+            debian_dir = os.path.join(rackhd_dir, "debian")
+            version_dict = self._generate_version_dict()
+            self._update_dependency(debian_dir, version_dict)
+        except Exception, e:
+            print "Failed to update RackHD/debian/control due to {0}".format(e)
+            sys.exit(1)
+
+        
+def parse_command_line(args):
+    """
+    Parse script arguments.
+    :return: Parsed args for assignment
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--manifest-repo-dir",
+                        required=True,
+                        help="the url of repository manifest",
+                        action="store")
+    
+    parser.add_argument("--builddir",
+                        required=True,
+                        help="destination for checked out repositories",
+                        action="store")
+
+    parser.add_argument("--git-credential",
+                        required=True,
+                        help="Git credentials for CI services",
+                        action="append")
+
+    parser.add_argument("--is-official-release",
+                        default=False,
+                        help="whether this release is official",
+                        action="store_true")
+
+    parsed_args = parser.parse_args(args)
+    return parsed_args
+
+def main():
+    # Parse arguments
+    args = parse_command_line(sys.argv[1:])
```

Comment:
Do you mean that main() should first checkout all the repositories according to manifest?